pkg/cmd/bazci/bazci.go

```diff
@@ -213,8 +214,9 @@ func (s *monitorBuildServer) handleBuildEvent(
 				if summary != nil && summary.ShardCount > 1 {
 					outputDir = filepath.Join(outputDir, fmt.Sprintf("shard_%d_of_%d", testResult.shard, summary.ShardCount))
 				}
-				if testResult.attempt > 1 {
-					outputDir = filepath.Join(outputDir, fmt.Sprintf("attempt_%d", testResult.attempt))
+				// Only upload results of first successful attempt or last attempt.
+				if testResult.attempt < maxTestAttempts && testResult.testResult.Status == bes.TestStatus_FAILED {
```

Comment:
right.. what do you think about adding a flag `flaky_unit_test_attempts` to `bazci` and when it's passed we pass `flaky_test_attempts` to `bazel`?

Reply:
and we would add to the condition above: `&& flaky_unit_test_attempts was passed` + make `maxTestAttempts be the value of the flag ofcourse

Reply:
That could work but it does require us to make sure we use the flag properly and only ever pass it to `bazci` rather than to `bazel`.

There is another way to look at it that requires no special handling for this in `bazci`. I think what you want to be doing is looking at the BEP results from Bazel and manually de-duping the results. For example, if a test was re-tried, you're going to get multiple `testResult`s that had different `attempt` numbers, and in this case you should be selecting the one with the greatest `attempt` number. I *think* if a test is re-tried then all the retries will be packed together in a single `BuildEventId_TestSummary` and you should be able to traverse that in-memory, but you should verify this.